components/workspacekit/pkg/lift/lift.go

```diff
@@ -0,0 +1,169 @@
+package lift
+
+// Lift can execute commands in a namespace context and return the stdin/out/err FDs
+// to the caller. This allows us to lift commands into ring1, akin to `docker exec`.
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	DefaultSocketPath = "/tmp/workspacekit-lift.socket"
+
+	// must not clash with the socket control message types:
+	//    https://github.com/torvalds/linux/blob/master/include/linux/socket.h#L163
+	msgTypeLiftRequest = 0xCA01
+)
+
+type LiftRequest struct {
+	Command []string `json:"command"`
+}
+
+func ServeLift(socket string) error {
+	skt, err := net.Listen("unix", socket)
+	if err != nil {
+		return err
+	}
+	defer skt.Close()
+
+	for {
+		conn, err := skt.Accept()
+		if err != nil {
+			log.WithError(err).Error("cannot accept lift connection")
+			continue
+		}
+
+		go func() {
```

Comment:
Is this goroutine cleaned up somehow? Or does it not matter because then the whole process is terminated/killed?